pkg/cloud/aws/services/ec2/instances.go

```diff
@@ -205,7 +207,7 @@ func (s *Service) createInstance(machine *actuators.MachineScope, bootstrapToken
 				ServiceSubnet:     s.scope.Cluster.Spec.ClusterNetwork.Services.CIDRBlocks[0],
 				ServiceDomain:     s.scope.Cluster.Spec.ClusterNetwork.ServiceDomain,
 				KubernetesVersion: machine.Machine.Spec.Versions.ControlPlane,
-			})
+			}, v1beta1.InitConfiguration{})
```

Comment:
Need to pass the user specified config int this function instead of an empty config

Reply:
Not related to this line, but I see that only the controlplane user data is being enriched with the JoinConfiguration, the node user data should also be as well.

Reply:
yeah, absolutely